src/charm.py

```diff
@@ -96,12 +94,25 @@ def _on_config_changed(self, _):
         self._stored.config["mirror-list"] = self.model.config[
             "mirror-list"
         ].splitlines()
-        self._render_config(self._stored.config)
-        if (
-            "cron-schedule" in self._stored.config
-            and self._stored.config["cron-schedule"] != "None"  # noqa: W503
-        ):
-            self._setup_cron_job(self._stored.config)
+
+        # use change set to support single dispatch of a config change.
+        template_change_set = set(
+            {
+                "base-path",
+                "architecture",
+                "threads",
+                "http_proxy",
+                "https_proxy",
+                "mirror-list",
+            }
+        )
+        if len(change_set & template_change_set) > 0:
+            self._render_config(self._stored.config)
+        if "cron-schedule" in change_set:
+            if self._stored.config["cron-schedule"] == "None":
```

Comment:
Is this a good practice to use `"None"` instead of `""`? At least for me it would be more intuitive to set it as empty string.
What do you think?

Reply:
Yes, that's true.